### PR TITLE
Feature/repo metrics ui

### PR DESCRIPTION
**Pull request by bparish628**
Wanted to get this out there for you to see. It is a very rough display stats page. The tests are a bit wonky, so I'll be adding more/adding fixing them with a new way.

### CODE REVIEW

File: repo_health/index/static/client/src/components/search/search.html [hunk 1]
```diff
@@ -1,15 +1,34 @@
...
-      </div>
-    </form>
+<div class="container">
+  <md-content class="md-padding" layout-xs="column" layout="row" ng-if="$ctrl.error">
+    <div flex-xs flex-gt-xs layout="column">
+      <md-card md-theme="error">
+        <md-card-title>
+          <md-card-title-text>
+            <span class="md-headline">Error</span>
+            <span class="md-subhead">This repo does not exist</span>
```
Comment (jakeharding) [inline repo_health/index/static/client/src/components/search/search.html]: Should we reference the `$ctrl.error` text here?

**Review by jakeharding — COMMENTED** [review]
This is good stuff. I really like the ES6 and the angluar material stuff.  It really makes the code easier to read.  We do need to get the tests figured out before we can merge though.

File: repo_health/index/static/client/src/components/repo-details/repo-details.spec.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+      it('should setup RepoDetailsService, $state, $stateParams', () => {
+        expect(controller.RepoDetailsService).toBeOk;
+        expect(controller.$state).toBeOk;
+        expect(controller.$stateParams).toBeOk;
+      });
+    });
+
+    describe('getStats', () => {
+      it('should make a call to service', async () => {
+        spyOn(controller.RepoDetailsService, 'getStats').and.returnValue(Promise.resolve({ name: 'cakephp' }));
```
Comment (jakeharding) [inline repo_health/index/static/client/src/components/repo-details/repo-details.spec.js]: This test was failing on my local but returning the object directly without the `Promise.resolve` made it pass.

**Comment by bparish628** [comment]
@jakeharding Hey I updated everything with correct tests. I found out why nothing was working though! It was because Jasmines assertions are actually functions and not just variables (I'm used to used chai).

File: repo_health/index/static/client/src/components/repo-details/repo-details.spec.js [hunk 3]
```diff
@@ -26,35 +29,83 @@ describe('Repo Details', () => {
...
       $componentController = $injector.get('$componentController');
       $rootScope = $injector.get('$rootScope');
+      $q = $injector.get('$q');
       controller = $componentController('repoDetails');
     }));
 
     describe('constructor', () => {
       it('should setup the controller', () => {
-        expect(controller).toBeDefined;
+        expect(controller).toBeDefined();
```
Comment (jakeharding) [inline repo_health/index/static/client/src/components/repo-details/repo-details.spec.js]: Ha! I wondered why you didn't call these as methods.  I figured it was some crazy new ES6 thing. Face palm!

File: repo_health/index/static/client/src/components/repo-details/repo-details.spec.js [hunk 4]
```diff
@@ -26,35 +29,83 @@ describe('Repo Details', () => {
...
-        expect(controller.RepoDetailsService.getStats).toHaveBeenCalled;
+        expect(controller.loadingRepo).toBeFalsy();
+        expect(controller.details).toEqual({ name: 'cakephp' });
+        expect(controller.RepoDetailsService.getStats).toHaveBeenCalled();
+      });
+    });
+  });
+
+    describe('RepoDetailsService', () => {
+    let RepoDetailsService;
```
Comment (jakeharding) [inline repo_health/index/static/client/src/components/repo-details/repo-details.spec.js]: Could you indent this block for readability?

**Review by jakeharding — APPROVED** [review]

File: repo_health/index/static/client/src/components/search/search.spec.js [hunk 5]
```diff
@@ -10,27 +10,69 @@
...
     describe('getStats', () => {
-      it('should return the stats', () => {
-        //Add tests when function is implemented
+      let getStats;
+
+      it('should not make a call if url is invalid', () => {
+        spyOn(controller.RepoDetailsService, 'getStats');
+        controller.githubUrl = 'This not a url';
+        controller.getStats();
+        expect(controller.loadingRepo).toBeTrue;
```
Comment (jakeharding) [inline repo_health/index/static/client/src/components/search/search.spec.js:56]: Pulled in the changes and rebased my working branch and found this when I looking through the files.  Should be `toBeTruthy()`. We can get it in the next PR.

Comment (bparish628) [reply repo_health/index/static/client/src/components/search/search.spec.js:56]: Nooo. I thought I got them all :cry:

Comment (jakeharding) [reply repo_health/index/static/client/src/components/search/search.spec.js:56]: 😁  It was a sneaky one!

**Pull request merged**